File: app/src/main/java/com/dhsdevelopments/potato/clientapi/notifications/StateUpdateNotification.java
package com.dhsdevelopments.potato.clientapi.notifications;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class StateUpdateNotification extends PotatoNotification
{
    @SerializedName( "add-type" )
    public String addType;

    @SerializedName( "user" )
    public String userStateUser;

    @SerializedName( "users" )
    public List<UserStateUpdateUser> userStateSyncMembers;

    @SerializedName( "channel" )
    public String channel;

    @Override
    public String toString() {
        return "StateUpdateNotification[" +
                       "addType='" + addType + '\'' +
                       ", userStateUser='" + userStateUser + '\'' +
                       ", userStateSyncMembers=" + userStateSyncMembers +
                       ", channel='" + channel + '\'' +
                       "] " + super.toString();
    }
}

File: app/src/main/java/com/dhsdevelopments/potato/clientapi/notifications/StateUpdateNotification.kt
package com.dhsdevelopments.potato.clientapi.notifications

import com.google.gson.annotations.SerializedName

class StateUpdateNotification : PotatoNotification() {
    @SerializedName("add-type")
    lateinit var addType: String

    @SerializedName("user")
    var userStateUser: String? = null

    @SerializedName("users")
    var userStateSyncMembers: List<UserStateUpdateUser>? = null

    @SerializedName("channel")
    lateinit var channel: String

    override fun toString(): String {
        return "StateUpdateNotification[" +
                "addType='" + addType + '\'' +
                ", userStateUser='" + userStateUser + '\'' +
                ", userStateSyncMembers=" + userStateSyncMembers +
                ", channel='" + channel + '\'' +
                "] " + super.toString()
    }
}
